lib/jsonapi/view.ex

```diff
@@ -91,7 +91,19 @@ defmodule JSONAPI.View do
       end
 
       def attributes(data, conn) do
-        visible_fields(data)
+        visible_fields = fields() -- hidden()
+
+        Enum.reduce(visible_fields, %{}, fn field, intermediate_map ->
```

Comment:
Why `reduce/3` vs `map/2`?

Reply:
This was to ensure the return type was a Map, which is what `Map.take` was doing before.